3rd/just_file/include/just/file.hpp

```diff
@@ -26,33 +28,56 @@ namespace just
 
     namespace impl
     {
-      inline void write(
-        const std::string& filename_,
-        const char* content_,
-        std::streamsize len_
-      )
+      inline void throw_errno(const std::string& action_)
       {
-        std::ofstream f(filename_.c_str());
-        if (f)
+        throw error(action_ + ": " + std::strerror(errno));
+      }
+
+      class output_file
+      {
+      public:
+        explicit output_file(const std::string& filename_) :
+          _filename(filename_),
+          _f(std::fopen(filename_.c_str(), "wb"))
         {
-          if (!f.write(content_, len_))
+          if (!_f)
           {
-            throw error("Failed to write to file " + filename_);
+            throw_errno("Error creating file " + filename_);
           }
         }
-        else
+
+        ~output_file()
         {
-          throw error("Failed to create file " + filename_);
+          fclose(_f);
         }
-      }
+
+        void write(const char* content_, std::size_t len_)
+        {
+          if (len_ > 0 && fwrite(content_, len_, 1, _f) < 1)
+          {
+            throw_errno("Error writing to " + _filename);
+          }
+        }
+      private:
+        std::string _filename;
+        std::FILE* _f;
```

Comment:
Just curious. Why not `std::ofstream`?

Reply:
The only benefit of using `std::ofstream` would be resource management, so I rather used the lower-level API without the overhead of streams.